### PR TITLE
Heirloom-devutils: yacc and lex

### PR DESCRIPTION
**Pull request by stikonas**
Heirloom `yacc` and `lex` will be needed to bootstrap `flex` and `bison`.

### CODE REVIEW

File: sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Part 1: Heirloom yacc
+cd ${pkg}/yacc
+
+# Prepare and patch
+patch -Np0 -i ../../../patches/yacc-wchar.patch
+patch -Np0 -i ../../../patches/fakewchar.patch
+
+# Build
+make -f Makefile.mk CC=tcc AR=tcc\ -ar CFLAGS+=-I.\ -DMAXPATHLEN=100\ -DEILSEQ=84\ -DMB_LEN_MAX=100\ -Dswprintf=sprintf\ -Dwcslen=strlen\ -Diswdigit=isdigit\ -Diswxdigit=isxdigit\ -Diswlower=islower\ -Diswupper=isupper\ -Diswspace=isspace\ -Dwcscmp=strcmp\ -Dtowlower=tolower\ -Diswalpha=isalpha\ -Diswalnum=isalnum\ -Dungetwc=ungetc\ -Dwcscpy=strcpy\ -Dgetwc=getc\ -Dputwc=putc LDFLAGS=-lgetopt RANLIB=true
```
Comment (fosslinux) [inline sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem:20]: Split on `\`?

**Review by fosslinux — CHANGES_REQUESTED** [review]
A readme bit would also be neat.

File: sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem (hunk 1 above)
Comment (stikonas) [reply sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem:20]: Ok, I can split.

As for `README`, I thought we should finish at least both heirloom `lex` and `yacc` and then describe both in one paragraph.

If you want I can write now, but I thought it's better to wait until we are done with `lex`.

Comment (stikonas) [reply sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem:20]: Hmm, splitting lines in kaem is a bit tricky. I tried to add `\` at the end of the line and no spaces on the next line but it still didn't work...

Comment (fosslinux) [reply sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem:20]: Lol forgot it was kaem. Please just remove the `\` then.

**Comment by fosslinux** [comment]
regarding README, yeah, we can just describe both in one para later.

**Comment by stikonas** [comment]
Ok, I think `yacc` hits some mes libc bug because when building on normal system with the same version of tinycc but with glibc it works.

File: sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem (hunk 1 above)
Comment (stikonas) [reply sysa/heirloom-devtools-070527/heirloom-devtools-070527.kaem]: Agreed on IRC to keep it as it is.

**Comment by stikonas** [comment]
Added lex too although at the moment it fails to compile.

Generated parse.c file contains tokens like

`# define C 257`
instead of
`# define CHAR 257`

and also
```
-static YYCONST yytabelem yyact[]={
+static YYCONST yytabelem y[]={
```

**Comment by stikonas** [comment]
`yacc` is ready to go. Lex will be done in a follow-on PR.

**Comment by stikonas** [comment]
Lex is now working too.

**Pull request merged**